### DISCUSSION TITLE
动态路由鉴权如何写呀？没有看到文档

### DISCUSSION THREAD

**Discussion by lqliqi** [umijs/umi#9923] [Q&A]
静态路由写法是  {
      icon: 'AliwangwangOutlined',
      name: '权限演示',
      path: '/access',
      component: './Access',
      wrappers: ['@/wrappers/Auth'],
    }, 直接在路由项里面配置了一个 wrappers 字段，指向包装路由的路由，但是写了动态路由后，包装组件的写法有点不会写？
我参考了 element:  React.createElement(
            require('@/pages/SystemSet/AccountManage/EddAccount').default,
          ) 组件的写法： obj.wrappers = [
            React.createElement(require('@/wrappers/Auth').default),
          ];  但是没有效果，求大佬解答

**Comment by fz6m** [comment]
需要维护一份全部路由的映射，可以参考这个 demo ，原理差不多：https://github.com/umijs/umi/discussions/9717#discussioncomment-4078053

如果是在 wrapper / layout 里鉴权，发现没有权 redirect 到主页就行了，不需要动态路由，动态路由是指服务端下发的。
